### PR TITLE
fix guess_start_dir in case we're already in start_dir

### PR DESCRIPTION
**Pull request by boegel**
follow-up for #1413, since that was a non-backwards compatible change (I realised that the second after hitting the shiny green `Merge` button)

@wpoely86: can you review and maybe look into an enhanced unit test for this?

### CODE REVIEW

File: easybuild/framework/easyblock.py [hunk 1]
```diff
@@ -1136,19 +1136,28 @@ def guess_start_dir(self):
...
+                topdir = self.builddir
+
+            abs_start_dir = os.path.join(topdir, start_dir)
+            if topdir.endswith(start_dir) and not os.path.exists(abs_start_dir):
+                self.cfg['start_dir'] = topdir
+            else:
+                if os.path.exists(abs_start_dir):
+                    self.cfg['start_dir'] = abs_start_dir
+                else:
+                    raise EasyBuildError("Specified start dir %s does not exist", abs_start_dir)
```
Comment (wpoely86) [inline easybuild/framework/easyblock.py]: print `self.cfg['start_dir']` here as that is the dir that is actually specified?

**Comment by wpoely86** [comment]
Can this not be done simpler by letting the extract_step not change the directory but simple setting the `finaldir`?

**Comment by boegel** [comment]
@wpoely86: you're working on a test for this? PR pending?

**Comment by wpoely86** [comment]
@boegel I've sent you a PR with it: boegel/easybuild-framework#10

**Comment by wpoely86** [comment]
Btw, it can not be done simpler. This is probably the best solution.

File: easybuild/framework/easyblock.py (hunk 1 above)
Comment (boegel) [inline easybuild/framework/easyblock.py]: no, this error applies to the failing `if os.path.exists(abs_start_dir)`, but I'll include an extra debug log message printing the `start_dir` we end up with

**Comment by boegel** [comment]
@wpoely86: I fleshed out the test you wrote a bit more, please rererereview?

Should be good to go in otherwise, imho.

**Comment by wpoely86** [comment]
looks good

File: easybuild/tools/filetools.py [hunk 2]
```diff
@@ -840,6 +840,31 @@ def expand_glob_paths(glob_paths):
...
+    # strip path1 for use in comparisons
+    path1s = path1.strip(os.path.sep)
+
+    # init part2 head/tail/parts
+    path2_head = path2.strip(os.path.sep)
+    path2_tail = ''
+    path2_parts = path2.split(os.path.sep)
+
+    while path2_parts and not path1s.endswith(path2_head):
+        path2_tail = os.path.join(path2_parts.pop(-1), path2_tail)
```
Comment (wpoely86) [inline easybuild/tools/filetools.py]: the `-1` is not needed?

File: test/framework/filetools.py [hunk 3]
```diff
@@ -429,12 +429,26 @@ def test_multidiff(self):
...
+        self.assertEqual(ft.weld_paths('/foo/bar', 'foobar/baz'), '/foo/bar/foobar/baz/')
+        self.assertEqual(ft.weld_paths('foo', 'bar/'), 'foo/bar/')
+        self.assertEqual(ft.weld_paths('foo/', '/bar'), '/bar/')
+
+        # overlap is taken into account
+        self.assertEqual(ft.weld_paths('foo/bar', 'bar/baz'), 'foo/bar/baz/')
+        self.assertEqual(ft.weld_paths('foo/bar/baz', 'bar/baz'), 'foo/bar/baz/')
+        self.assertEqual(ft.weld_paths('foo/bar', 'foo/bar/baz'), 'foo/bar/baz/')
+        self.assertEqual(ft.weld_paths('foo/bar', 'foo/bar'), 'foo/bar/')
+        self.assertEqual(ft.weld_paths('/foo/bar', 'foo/bar'), '/foo/bar/')
```
Comment (wpoely86) [inline test/framework/filetools.py]: add two absolute paths with and without overlap?

**Comment by wpoely86** [comment]
lgtm

**Comment by boegel** [comment]
Going in, thanks for the review @wpoely86!

**Pull request merged**